Src/IronPython/Runtime/Generator.cs

```diff
@@ -24,6 +24,7 @@ public sealed class PythonGenerator : IEnumerator, IEnumerator<object>, ICodeFor
         private readonly PythonFunction _function;                  // the function which created the generator
         private readonly MutableTuple _data;                        // the closure data we need to pass into each iteration.  Item000 is the index, Item001 is the current value
         private readonly MutableTuple<int, object> _dataTuple;      // the tuple which has our current index and value
+        private Exception _exceptionContext;                        // the exception being handled in the generator itself
```

Comment:
Do I need to initialize this to the current exception at the start?

Reply:
Hmm, maybe? Not sure if this example relates...
```Python
def test4():
    def gen():
        try:
            raise Exception
        except:
            pass
        yield sys.exc_info()
    x = gen()
    try:
        raise Exception
    except:
        exc_info = sys.exc_info()
        assert next(x) == exc_info

test4()
```

Reply:
I did not expect that behavior -- we get the exception context from the first call to `next()`?

Reply:
I don't think this it's getting the exception from the first call to `next()`. It's basically whatever exception is currently being handled:
```Python
def test5():
    def gen():
        yield sys.exc_info()
        try:
            raise Exception
        except:
            pass
        yield sys.exc_info()
    x = gen()
    assert next(x) == (None, None, None)
    try:
        raise Exception
    except:
        exc_info = sys.exc_info()
        assert next(x) == exc_info

test5()
```

Reply:
Ah I see -- it looks like the handling is mostly intuitive. One edge case we might need to look out for is these two:
```
def test6():
    def gen():
        try:
            raise Exception
        except:
            yield traceback.format_exc()
        try:
            raise Exception
        except:
            yield traceback.format_exc()
        
    x = gen()
    try:
        raise Exception('@Exception 1')
    except:
        assert '@Exception 1' in next(x)
    try:
        raise Exception('@Exception 2')
    except:
        res = next(x)
        assert '@Exception 1' in res
        assert '@Exception 2' not in res

test6()

def test7():
    def gen():
        try:
            raise Exception
        except:
            yield traceback.format_exc()
        yield "Hello"
        try:
            raise Exception
        except:
            yield traceback.format_exc()
        
    x = gen()
    try:
        raise Exception('@Exception 1')
    except:
        assert '@Exception 1' in next(x)
    next(x)
    try:
        raise Exception('@Exception 2')
    except:
        res = next(x)
        assert '@Exception 2' in res
        assert '@Exception 1' not in res

test7()
```

Reply:
But this probably just follows from the implementation of try/catch that we have.

Reply:
Hmm, test6 is unexpected. It passes with 3.4 but fails with 3.7.3. Might have been a bug in the CPython implementation?

Personally I would prefer to match 3.7 on this one, but I wouldn't go out of my way to do it at this point in time.

Reply:
I'm using 3.6 so it looks like it was probably changed in 3.7. I'll try to get the error inheritance to work first.